bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,277 @@
+import re
+from .._numbadeclare import _NumbaDeclareDecorator
+
+# This class is still to undergo some restructuring thus the docstring may not be complete.
+# The class will be fully realized after the Define has been Pythonized.
+class FunctionJitter:
+    """
+    Class to jit function using NumbaDeclare.
+    It takes a python callable (or expression), an RDF and optionally a column list and dictionary containing any other arguments for the callable.
+    It determines the function signature and Declares the function and returns the corresponding function call.
+
+    Attributes
+    ----------
+    rdf: RDataFrame object on which the function's column arguments are in
+    col_names: The list of columns in rdf 
+    func: The python callable that is to be jitted.
+    params: List of parameters of the above callable
+    func_args: Dict to store the value of each param
+    args_info: Information about the type and value of each param
+    func_sign: Contains the function signature
+    func_call: Contains the function call as a string.
+
+    Methods
+    -------
+    find_type(x): determines the type of the variable passed
+    find_function_params(func): determines the parameter list of the python callable passed
+    generate_func_args(col_list, extra_args): uses the user passed column_list and extra_args to determine the value and type of each parameter of the callable. Stores it in arg_info
+    find_function_signature(): Calculates arg_info containing type  and value of each parameter and stores it in arg_info
+    generate_function_call(): Uses the arg_info to create a NumbaDeclare compatible function signature and the corresponding function call.
+    jit_function(func, cols_list, extra_args): Uses the above functions to jit the function and return's its call   
+
+    eg:
+
```

Comment:
It would be better to have the docstrings of the methods within the methods themselves only, otherwise there's duplication and more maintenance in case of changes.

Reply:
Should I just list the method names or should I take them out entirely?